mountpoint-s3-client/src/s3_crt_client/get_object.rs

```diff
@@ -109,16 +139,26 @@ pub struct S3GetObjectRequest {
     finish_receiver: UnboundedReceiver<Result<GetBodyPart, Error>>,
     finished: bool,
     enable_backpressure: bool,
+    object_metadata_cb: Lazy<Result<ObjectMetadata, Canceled>, Receiver<ObjectMetadata>>,
     /// Next offset of the data to be polled from [poll_next]
     next_offset: u64,
     /// Upper bound of the current read window. When backpressure is enabled, [S3GetObjectRequest]
     /// can return data up to this offset *exclusively*.
     read_window_end_offset: u64,
 }
 
+#[cfg_attr(not(docsrs), async_trait)]
 impl GetObjectRequest for S3GetObjectRequest {
     type ClientError = S3RequestError;
 
+    async fn get_object_metadata(&mut self) -> Result<HashMap<String, String>, Self::ClientError> {
```

Comment:
does it have to be `mut`?

https://docs.rs/async-once-cell/latest/src/async_once_cell/lib.rs.html#1277

Reply:
Yes - otherwise the failure client doesn't compile as it then requires the `FailureGetRequest` to be `Sync`, which it isn't. Adding `mut` to the signature relaxes this requirement and I wasn't sure how to resolve the failure client issue, and I don't think reducing from`mut` to non-mutable later is a breaking change if we ever need to reduce it